pilot/pkg/xds/ecds.go

```diff
@@ -167,13 +167,11 @@ func referencedSecrets(proxy *model.Proxy, push *model.PushContext, resourceName
 	// TODO: we get the WasmPlugins here to get the secrets reference in order to decide whether ECDS push is needed,
 	//       and we will get it again at extension config build. Avoid getting it twice if this becomes a problem.
 	watched := sets.New(resourceNames...)
-	wasmPlugins := push.WasmPlugins(proxy)
+	wasmPlugins := push.WasmPluginsByName(proxy, core.ParseExtensionName(resourceNames))
```

Comment:
This is subtle extremely unsafe and a major security issue. Basically this function is saying who can access what secrets. This is making it so anyone can request any resource, even if its in another namespace/not selecting their workload/etc

Reply:
cc @hzxuzhonghu

Reply:
@adilmp1 since this fix needs further discussion, i would suggest you to remove the fix from the current PR, and skip the third test case pointing to the issue you have raised. Once the discussion concludes we can decide how to add a proper fix in a separate PR

Reply:
Actually without this fix, we are allowing referencing any secret from any namespace.  Vagulaly remeber we did talk about allow sharing a secret for all wasm image pull.

 BTW, this is fixing a bug https://github.com/istio/istio/issues/51288